text/0148-sampling-probability.md

```diff
@@ -0,0 +1,805 @@
+sampling is performed while processing a stream of events and each
+event is considered just once.  Sampling designs of this form are
+referred to as _sampling without replacement_.  Unless stated
+otherwise, "sampling" in telemetry collection always refers to
+sampling without replacement.
+
+After executing a given sampling design over a complete frame of data,
+the result is a set of selected sample events, each having known and
+non-zero inclusion probability.  There are several other quantities of
+interest, after calculating a sample from a sample frame.
+
+- *Sample size*: the number of events with non-zero inclusion probability
+- *True population total*: the exact number of events in the frame, which may be unknown
+- *Estimated population total*: the estimated number of events in the frame, which is computed from the sample.
+
+The sample size is always known after it is calculated, but the size
+may or may not be known ahead of time, depending on the design.
+Probabilistic sampling schemes require that the estimated population
+total equals the expected value of the true population total.
+
+#### Adjusted sample count
+
+Following the model above, every event defines the notion of an
+_adjusted count_.
+
+- _Adjusted count_ is zero if the event was not selected for the sample
+- _Adjusted count_ is the reciprocal of its inclusion probability, otherwise.
+
+The adjusted count of an event represents the expected contribution to
+the estimated population total of a sample frame represented by the
+individual event.
+
+The use of a reciprocal inclusion probability matches our intuition
+for probabilities.  Items selected with "one-out-of-N" probability of
+inclusion count for N each, approximately speaking.
+
+This intuition is backed up with statistics.  This equation is known
+as the Horvitz-Thompson estimator of the population total, a
+general-purpose statistical "estimator" that applies to all _without
+replacement_ sampling designs.
+
+Assuming sample data is correctly computed, the consumer of sample
+data can treat every sample event as though an identical copy of
+itself has occurred _adjusted count_ times.  Every sample event is
+representative for adjusted count many copies of itself.
+
+There is one essential requirement for this to work.  The selection
+procedure must be _statistically unbiased_, a term meaning that the
+process is required to give equal consideration to all possible
+outcomes.
+
+#### Sampling and variance
+
+The use of unbiased sampling outlined above makes it possible to
+estimate the population total for arbitrary subsets of the sample, as
+every individual sample has been independently assigned an adjusted
+count.
+
+There is a natural relationship between statistical bias and variance.
+Approximate counting comes with variance, a matter of fact which can
+be controlled for by the sample size.  Variance is unavoidable in an
+unbiased sample, but it vanishes when you have enough data.
```

Comment:
This sounds like there's some magic threshold for sample size. Maybe rephrase that variance diminishes with growing sample size.